commons/src/main/java/org/infinispan/commons/configuration/attributes/AttributeSet.java

```diff
@@ -269,6 +269,25 @@ public boolean equals(Object obj) {
       return true;
    }
 
+   public boolean matches(AttributeSet other) {
+      if (other.attributes.size() != attributes.size())
+         return false;
+      Iterator<Map.Entry<String, Attribute<?>>> i = attributes.entrySet().iterator();
+      while (i.hasNext()) {
```

Comment:
foreach instead of iterator?

Reply:
... I guess not...

Reply:
Well, I want to exit the loop fast if there is a non-match.

Reply:
not the `forEach()`, this "for-each"
```
for (Map.Entry<String, Attribute<?>> e : attributes.entrySet()) {
```